### PR TITLE
[MU3] Instruments recognition fix

### PR DESCRIPTION
**Pull request by RomanPudashkin**

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
This seems to result in Oboe to get missdetected as Piccolo Oboe

**Comment by Jojo-Schmitz** [comment]
and "guitar-nylon-treble-clef" as "guitar-steel-drums" and bass as bass-flute etc. in tst_mxml_io

File: libmscore/instrtemplate.cpp [hunk 1]
```diff
@@ -753,12 +753,32 @@ InstrumentTemplate* searchTemplateForMusicXmlId(const QString& mxmlId)
...
+
       for (InstrumentGroup* g : qAsConst(instrumentGroups)) {
             for (InstrumentTemplate* it : qAsConst(g->instrumentTemplates)) {
-                  for (const QString& name : nameList) {
-                        if (it->trackName == name ||
-                            it->longNames.contains(StaffName(name)) ||
-                            it->shortNames.contains(StaffName(name)))
+                  QString trackNameLower = it->trackName.toLower();
+                  QStringList longNames = staffNameListToLower(it->longNames);
+                  QStringList shortNames = staffNameListToLower(it->longNames);
```
Comment (Jojo-Schmitz) [inline libmscore/instrtemplate.cpp:776]: not `it->shortnames` here?

File: libmscore/instrtemplate.cpp [hunk 2]
```diff
@@ -782,17 +806,10 @@ InstrumentTemplate* searchTemplateForMidiProgram(int midiProgram, const bool use
...
-                            name.contains(it->longNames.value(0).name(), Qt::CaseInsensitive) ||
-                            name.contains(it->shortNames.value(0).name(), Qt::CaseInsensitive)) {
-                              return it;
-                              }
-                        }
-                  }
-            }
+      InstrumentTemplate* templ = searchTemplateForInstrNameList(nameDataList);
+      if (templ) {
+          return templ;
```
Comment (Jojo-Schmitz) [inline libmscore/instrtemplate.cpp:811]: and else?

File: libmscore/instrtemplate.cpp (hunk 1 above)
Comment (RomanPudashkin) [reply libmscore/instrtemplate.cpp]: Thank you! Fixed

**Comment by RomanPudashkin** [comment]
@Jojo-Schmitz 

For some reason, those tests fail even without my changes. It seems that something has been broken in the 3.6.2_backend branch

**Comment by Jojo-Schmitz** [comment]
On my branch the tests passed before your changes but not after

File: libmscore/instrtemplate.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply libmscore/instrtemplate.cpp]: That doesn't fix the mtests though

**Review by Eism — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
FWIW: I can approve this too, the mtests work again, see https://github.com/Jojo-Schmitz/MuseScore/runs/3359680621

**Pull request merged**